pkg/http/core/search.go

```diff
@@ -1,11 +1,133 @@
 package core
 
 import (
+	"errors"
+	"fmt"
 	"net/http"
+	"strconv"
+	"strings"
 
+	clouddriver "github.com/billiford/go-clouddriver/pkg"
+	"github.com/billiford/go-clouddriver/pkg/sql"
 	"github.com/gin-gonic/gin"
 )
 
+type SearchResponse []Page
+
+type Page struct {
+	PageNumber   int          `json:"pageNumber"`
+	PageSize     int          `json:"pageSize"`
+	Query        string       `json:"query"`
+	Results      []PageResult `json:"results"`
+	TotalMatches int          `json:"totalMatches"`
+	// Platform     string       `json:"platform"`
+}
+
+type PageResult struct {
+	Account        string `json:"account"`
+	Group          string `json:"group"`
+	KubernetesKind string `json:"kubernetesKind"`
+	Name           string `json:"name"`
+	Namespace      string `json:"namespace"`
+	Provider       string `json:"provider"`
+	Region         string `json:"region"`
+	Type           string `json:"type"`
+	Application    string `json:"application,omitempty"`
+	Cluster        string `json:"cluster,omitempty"`
+}
+
+// Example response
+//
+// [
+//   {
+//     "pageNumber": 1,
+//     "pageSize": 500,
+//     "platform": "aws",
+//     "query": "spinnaker",
+//     "results": [
+//       {
+//         "account": "gke_github-replication-sandbox_us-central1_sandbox-us-central1-dev",
+//         "group": "deployment",
+//         "kubernetesKind": "deployment",
+//         "name": "deployment spin-fiat",
+//         "namespace": "spinnaker",
+//         "provider": "kubernetes",
+//         "region": "spinnaker",
+//         "type": "serverGroupManagers"
+//       },
+//       {
+//         "account": "gke_github-replication-sandbox_us-central1_sandbox-us-central1-dev",
+//         "group": "deployment",
+//         "kubernetesKind": "deployment",
+//         "name": "deployment spin-front50",
+//         "namespace": "spinnaker",
+//         "provider": "kubernetes",
+//         "region": "spinnaker",
+//         "type": "serverGroupManagers"
+//       }
+// 	   ],
+//     "totalMatches": 2
+//   }
+// ]
 func Search(c *gin.Context) {
-	c.JSON(http.StatusOK, []string{})
+	sc := sql.Instance(c)
+	pageSize, _ := strconv.Atoi(c.Query("pageSize"))
+	namespace := c.Query("q")
+	// The "type" query param is the kubernetes kind.
+	kind := c.Query("type")
+	// Get all accounts the user has access to.
+	accounts := strings.Split(c.GetHeader("X-Spinnaker-Accounts"), ",")
+
+	if kind == "" || namespace == "" {
+		clouddriver.WriteError(c, http.StatusBadRequest, errors.New("must provide query params 'q' and 'type'"))
```

Comment:
I don't think that it's clear that q is for namespace. Maybe you could say `errors.New("must provide query params 'q' to specify the namespace and 'type' to specify the kind")`